crates/bevy_core_widgets/src/core_scrollbar.rs

```diff
@@ -0,0 +1,310 @@
+use bevy_app::{App, Plugin, PostUpdate};
+use bevy_ecs::{
+    component::Component,
+    entity::Entity,
+    hierarchy::{ChildOf, Children},
+    observer::On,
+    query::{With, Without},
+    system::{Query, Res},
+};
+use bevy_math::Vec2;
+use bevy_picking::events::{Drag, DragEnd, DragStart, Pointer, Press};
+use bevy_ui::{
+    ComputedNode, ComputedNodeTarget, Node, ScrollPosition, UiGlobalTransform, UiScale, Val,
+};
+
+/// Used to select the orientation of the scrollbar.
+#[derive(Debug, Default, Clone, Copy, PartialEq)]
+pub enum Orientation {
+    /// Horizontal orientation (stretching from left to right)
+    Horizontal,
+    /// Vertical orientation (stretching from top to bottom)
+    #[default]
+    Vertical,
+}
+
+/// A headless scrollbar widget, which can be used to build custom scrollbars.
+///
+/// Scrollbars operate differently than the other core widgets in a number of respects.
+///
+/// Unlike sliders, scrollbars don't have an `AccessibilityNode` component, nor can they have
+/// keyboard focus. This is because scrollbars are usually used in conjunction with a scrollable
+/// container, which is itself accessible and focusable. This also means that scrollbars don't
+/// accept keyboard events, which is also the responsibility of the scrollable container.
+///
+/// Scrollbars don't emit notification events; instead they modify the scroll position of the
+/// target entity directly.
+///
+/// A scrollbar can have any number of child entities, but one entity must be the scrollbar
+/// thumb, which is marked with the [`CoreScrollbarThumb`] component. Other children are ignored.
+/// The core scrollbar will directly update the position and size of this entity; the application
+/// is free to set any other style properties as desired.
+///
+/// The application is free to position the scrollbars relative to the scrolling container however
+/// it wants: it can overlay them on top of the scrolling content, or use a grid layout to displace
+/// the content to make room for the scrollbars.
+#[derive(Component, Debug)]
+#[require(ScrollbarDragState)]
+pub struct CoreScrollbar {
+    /// Entity being scrolled.
+    pub target: Entity,
+    /// Whether the scrollbar is vertical or horizontal.
+    pub orientation: Orientation,
+    /// Minimum size of the scrollbar thumb, in pixel units.
+    pub min_thumb_size: f32,
+}
+
+/// Marker component to indicate that the entity is a scrollbar thumb. This should be a child
+/// of the scrollbar entity.
+#[derive(Component, Debug)]
+pub struct CoreScrollbarThumb;
+
+impl CoreScrollbar {
+    /// Construct a new scrollbar.
+    ///
+    /// # Arguments
+    ///
+    /// * `target` - The scrollable entity that this scrollbar will control.
+    /// * `orientation` - The orientation of the scrollbar (horizontal or vertical).
+    /// * `min_thumb_size` - The minimum size of the scrollbar's thumb, in pixels.
+    pub fn new(target: Entity, orientation: Orientation, min_thumb_size: f32) -> Self {
```

Comment:
`min_thumb_size` feels like a weird thing to put on the `CoreScrollbar` component. How painful would it be to move that to `CoreScrollbarThumb` instead?

Reply:
Hmmm, it's used in a lot of calculations. Let me think about it.